shm/inc/Island.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+class Island {
+public:
+    class Coordinates {
+    public:
+        Coordinates(int, int);
+        bool operator==(const Coordinates&);
+    private:
+        const int positionX_;
+        const int positionY_;
+    };
+
+    Coordinates getCoordinates() const { return position_; }
+    
+private:
+    const Coordinates position_;
+};
```

Comment:
New line at the end of file. Not really needed in modern compilers, but removes this red warning circle here.